src/test/tv/codely/scala_http_api/module/user/UserUnitTestCase.scala

```diff
@@ -0,0 +1,13 @@
+package tv.codely.scala_http_api.module.user
+
+import tv.codely.scala_http_api.module.UnitTestCase
+import tv.codely.scala_http_api.module.user.domain.{User, UserRepository}
+
+protected[user] trait UserUnitTestCase extends UnitTestCase {
+  protected val repository: UserRepository = mock[UserRepository]
```

Comment:
we could define it as private in order to force to use methods like the `shouldSearchAllUsers` one instead of directly using the `repository`

Reply:
Meh, we need it accessible from the child classes (test cases) because we need it in order to instantiate the use cases:
```scala
final class UsersSearcherTest extends UserUnitTestCase {
  private val searcher = new UsersSearcher(repository)
  // ...
```

Sorry for the confusion :/